tracing-journald/src/socket.rs

```diff
@@ -31,14 +33,41 @@ fn cmsg_buffer_size_for_one_fd() {
     assert_cmsg_bufsize()
 }
 
-pub fn send_one_fd(socket: &UnixDatagram, fd: RawFd) -> Result<usize> {
+fn create_sockaddr_un_for_path<P: AsRef<Path>>(path: P) -> Result<sockaddr_un> {
+    let mut addr: sockaddr_un = unsafe { std::mem::zeroed() };
+    addr.sun_family = AF_UNIX as _;
+
+    let path_bytes = path.as_ref().as_os_str().as_bytes();
+    // We need to take the final NULL byte into account.
+    if path_bytes.len() < addr.sun_path.len() - 1 {
+        unsafe {
+            std::ptr::copy_nonoverlapping(
+                path_bytes.as_ptr(),
+                addr.sun_path.as_mut_ptr() as *mut u8,
+                path_bytes.len(),
+            );
+        }
+        addr.sun_path[addr.sun_path.len() - 1] = 0;
+        Ok(addr)
+    } else {
+        Err(Error::from_raw_os_error(ENAMETOOLONG))
+    }
+}
+
+pub fn send_one_fd_to<P: AsRef<Path>>(socket: &UnixDatagram, fd: RawFd, path: P) -> Result<usize> {
     assert_cmsg_bufsize();
 
     let mut cmsg_buffer = AlignedBuffer {
         buffer: ([0u8; CMSG_BUFSIZE]),
     };
     let mut msg: msghdr = unsafe { zeroed() };
 
+    let mut addr = create_sockaddr_un_for_path(path)?;
```

Comment:
Could we replace this entire function with a constant? [`sockaddr_un`](https://docs.rs/libc/latest/libc/struct.sockaddr_un.html) isn't a complicated structure.

Reply:
I've inlined it.